adapter/internal/oasparser/model/mgw_swagger.go

```diff
@@ -325,6 +325,9 @@ func (endpoint *Endpoint) validateEndpoint() error {
 	if len(endpoint.ServiceDiscoveryString) > 0 {
 		return nil
 	}
+	if endpoint.Port > 65535 {
```

Comment:
Shouldn't we check the range 0-65535

Reply:
The port number is initially parsed as uint. Hence, if a negative value is provided, it'll be pared to 0 anyway.

Reply:
`strconv.ParseUint("-111", 10, 32)` ==> 0

Reply:
if negative value is provided then will it set the port as zero in envoy config, because we only check port is less than 65535

Reply:
FIxed https://github.com/wso2/product-microgateway/pull/1957/commits/2737ef833ec7e623f99f3917fa161a6b68c714e4